snuba/utils/streams/processing/strategies/streaming/transform.py

```diff
@@ -280,8 +278,9 @@ def __init__(
         max_batch_time: float,
         input_block_size: int,
         output_block_size: int,
-        metrics: MetricsBackend,
+        metrics: Metrics = DummyMetricsBackend,
     ) -> None:
+        assert isinstance(metrics, Metrics)
```

Comment:
Why is this needed ?

Reply:
It's not really needed, but I figured it's nicer to error here if something is wrong with `metrics` than later when .gauge(), .timing() or .increment() are used.

Reply:
Then could you please add a comment on the reason why the `isinstance` call is there ?

Reply:
done